test/puppetlabs/puppetdb/cli/services_test.clj

```diff
@@ -493,27 +476,38 @@
                             (let [result (f first?)]
                               (.await after-gc)
                               result))
-          log (atom [])]
-      (with-redefs [svcs/invoke-periodic-gc invoke-periodic]
+          log (atom [])
+          current-time (now)
+          six-days (.minus current-time (time/parse-period "6d"))
+          seven-days (.minus current-time (time/parse-period "7d"))
+          basic-date1 (time/unparse (time/formatters :basic-date) six-days)
+          basic-date2 (time/unparse (time/formatters :basic-date) seven-days)
+          iso_date1 (time/unparse (time/formatters :date-hour-minute-second) six-days)
+          iso_date2 (time/unparse (time/formatters :date-hour-minute-second) seven-days)
+          ago-mock (fn [_]
+                    (.minus current-time (time/parse-period "2d")))]
+      (with-redefs [svcs/invoke-periodic-gc invoke-periodic
+                    time/ago ago-mock]
```

Comment:
I think we should be able to avoid updating the timestamps on the reports in this test if we redef the `scf/storage/resource-events-expired?` func that's added in this pr. Redefining that func to `(constantly false)` allowed me to run the tests using the older timestamps. I was still seeing some transient failures due to the expected postgres error not always getting delivered from the future, but that appears to be unrelated to your changes.